### PR TITLE
Created package.json file & Updated Read Me

### PR DESCRIPTION
**Pull request by chalkers**
Added a package.json file to help handle JavaScript dependancies better.

Updated readme for the Espruino source requirement.

Included instructions for using npm as a task manager. i.e. `npm run build` & `npm start`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,10 +1,11 @@
 <!--- Copyright (c) 2013 Gordon Williams, Pur3 Ltd. See the file LICENSE for copying permission. -->
-EspruinoDocs
-============
+# EspruinoDocs
```
Comment (gfwilliams) [inline README.md]: Why did you change all these? Some of us like underlining the titles ;)

File: README.md [hunk 2]
```diff
@@ -16,76 +17,100 @@ This is basically [GitHub Markdown](https://help.github.com/articles/github-flav
...
+## Build Process
 
-Finally you need to install a node module for highlighting:
+### First Time:
+
+For the first time you can run:
 
 ```
-$ npm install marked highlight.js acorn express tern --save
+$ ./build.sh
```
Comment (gfwilliams) [inline README.md]: This might not be entirely right. For instance if you want your modules to be updated, or the pinout for each board to be updated, you'll still need build.sh I think?

**Comment by gfwilliams** [comment]
Thanks! Just merging this in. Made some comments though ^^^

File: README.md (hunk 1 above)
Comment (chalkers) [inline README.md]: I think they still are :)

File: README.md (hunk 2 above)
Comment (chalkers) [inline README.md]: Is that what this code does in the bash script?
```
python scripts/build_board_docs.py PICO_R1_3 pinout
mv boards/PICO_R1_3.html $DIR/html/boards
python scripts/build_board_docs.py ESPRUINOBOARD pinout
mv boards/ESPRUINOBOARD.html $DIR/html/boards
python scripts/build_board_docs.py ESP8266_BOARD pinout
mv boards/ESP8266_BOARD.html $DIR/html/boards
python scripts/build_board_docs.py EMW3165 pinout
mv boards/EMW3165.html $DIR/html/boards
```

What are the scenarios where you need to do this?

Also not too keen that that shell script randomly generates this folder - `WEBSITE=~/workspace/espruinowebsite` lol

Oh and there's this:
```
bash buildmodules.sh
```
Is that where you mean build modules.

So what's the main differences between bin/build.js and build.sh?

Comment (gfwilliams) [inline README.md]: Yeah, so the python generates something that looks like this: http://www.espruino.com/ReferenceSTM32VLDISCOVERY

And then that can get pulled into a markdown file like this: http://www.espruino.com/Pico

And yeah, `~/workspace/espruinowebsite` is where I have the source for the actual Espruino site. Simplest thing would be to just check for its existence and then refuse to copy modules if it's not there.

And yes `bash buildmodules.sh` checks for updated versions of each module and minifies any that have changed.

File: README.md (hunk 1 above)
Comment (gfwilliams) [inline README.md]: Yeah, they are the same - except *every other heading in the entire repo uses dashes*, so it feels a bit pointless to randomly change these to something different?

File: README.md (hunk 2 above)
Comment (chalkers) [inline README.md]: Just to clarify:

bin/build.js is purely for the documentation (minus the boards with pinouts)

build.sh has 3 tasks:
1. generate pinouts using python
2. build the production site
3. minifies the modules

If that's correct I'll either update the docs tomorrow or create some other tasks in JS and use npm to run those tasks :)

Comment (gfwilliams) [inline README.md]: Yes, sounds about right. Thanks!

I reckon `build.js` could call Python to generate the board docs directly (it'd be more sensible) - it was just me being lazy using bash for it.

The module minification is an important one, because it's very handy to be able call it separately if you're making changes and want to quickly see how the minified code comes out.
